ls1intum/Artemis
# `Programming exercises`: Limit the length of static code analysis issues to prevent failures when saving new results

Fixes an issue when messages in static code analysis issues are too long. Those are capped to 4.500 characters in the feedback message to prevent exceptions such as 

```
Jan 18 17:39:31 artemistest.ase.in.tum.de java[5201]: 2022-01-18 17:39:31.517  INFO 5201 --- [io-8080-exec-19] d.t.i.w.artemis.web.rest.ResultResource  : Artemis received a new result for build plan TESTCHMEBUDDYTEAM-TEAM1
Jan 18 17:39:31 artemistest.ase.in.tum.de java[5201]: 2022-01-18 17:39:31.561  WARN 5201 --- [io-8080-exec-19] o.z.problem.spring.common.AdviceTraits   : Bad Request: Validation failed for classes [de.tum.in.www1.artemis.domain.Feedback] during persist time for groups [javax.validation.groups.Default, ]
Jan 18 17:39:31 artemistest.ase.in.tum.de java[5201]: List of constraint violations:[
Jan 18 17:39:31 artemistest.ase.in.tum.de java[5201]:         ConstraintViolationImpl{interpolatedMessage='size must be between 0 and 5000', propertyPath=detailText, rootBeanClass=class de.tum.in.www1.artemis.domain.Feedback, messageTemplate='{javax.validation.constraints.Size.message}'}
Jan 18 17:39:31 artemistest.ase.in.tum.de java[5201]: ]
```

This is problematic because the whole result will then not be stored in the database and students just see a missing result and don't know what is happening.

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise with static code analysis enabled

1. Create the C programming exercise
2. Change the code and produce a very long static code analysis issue, see e.g. https://artemistest.ase.in.tum.de/course-management/2353/programming-exercises/13792/participations/333960/submissions
3. Commit and push the solution, wait some time, make sure the result is displayed in Artemis

### Review Progress
#### Code Review
#### Manual Tests
